springboot/src/main/java/org/mobilitydata/gtfsvalidator/springboot/GtfsValidatorController.java

```diff
@@ -0,0 +1,420 @@
+  }
+
+  /**
+   * Converts query parameters to {@code Argument}.
+   *
+   * @param output_base Base directory to store the outputs
+   * @param threads Number of threads to use
+   * @param country_code Country code of the feed, e.g., `nl`. It must be a two-letter country code
+   *     (ISO 3166-1 alpha-2)").
+   * @param url The fully qualified URL to download GTFS archive
+   * @param validation_report_name The name of the validation report including .json extension.
+   * @param system_error_report_name The name of the system error report including .json extension.
+   * @return the {@code Argument} instance generated from the query parameters passed.
+   */
+  private Arguments queryParametersToArguments(
+      String output_base,
+      String threads,
+      String country_code,
+      String url,
+      String validation_report_name,
+      String system_error_report_name) {
+    final String[] argv = {
+      "-o", output_base,
+      "-t", threads,
+      "-c", country_code,
+      "-u", url,
+      "-v", validation_report_name,
+      "-e", system_error_report_name
+    };
+    Arguments args = new Arguments();
+    JCommander jCommander = new JCommander(args);
+    jCommander.parse(argv);
+    return args;
+  }
+
+  private GoogleCredentials generateCredentials(String... keys) throws IOException {
+    HashMap<String, String> creds = new HashMap<>();
+    Gson gson = new Gson();
+    for (String key : keys) {
+      creds.put(key.toLowerCase(), System.getenv(key));
+    }
+    return GoogleCredentials.fromStream(
+        new ByteArrayInputStream(gson.toJson(creds).replace("\\n", "\n").getBytes()));
+  }
+
+  /**
+   * Pushes validation report to Google Cloud Storage. Returns the {@code HttpStatus} of the
+   * validation report storage process. Requires authentication to be set prior execution i.e. the
+   * following environment variables have to be defined:
+   *
+   * <ul>
+   *   <li>TYPE
+   *   <li>PROJECT_ID
+   *   <li>PRIVATE_KEY_ID
+   *   <li>PRIVATE_KEY
+   *   <li>CLIENT_EMAIL
+   *   <li>CLIENT_ID
+   *   <li>AUTH_URI
+   *   <li>TOKEN_URI
+   *   <li>AUTH_PROVIDER_X509_CERT_URL
+   *   <li>CLIENT_X509_CERT_URL
+   *   <li>VALIDATION_REPORT_BUCKET
+   * </ul>
+   *
+   * <p>Please check:
+   *
+   * <ul>
+   *   <li><a href=https://cloud.google.com/storage/docs/naming-buckets>this documentation for
+   *       bucket naming guidelines</a>
+   *   <li><a
+   *       href=https://cloud.google.com/docs/authentication/getting-started#setting_the_environment_variable>this
+   *       documentation for authentication procedure</a>
+   * </ul>
+   *
+   * @param commitSha the commit SHA
+   * @param datasetId the id of the dataset
+   * @param args the {@code Argument} generated from the query parameters
+   * @return the {@code HttpStatus} of the validation report storage process
+   */
+  private HttpStatus pushValidationReportToCloudStorage(
```

Comment:
This method throws IOException, which would then also be thrown by `run()`, which I assume would crash the thread handling the incoming HTTP request. Is this what you want? Seems like IOExceptions here should be handled and response with status code returned to client.

Reply:
My bad - I forgot to push code I had on my laptop. https://github.com/MobilityData/gtfs-validator/pull/933/commits/3bfbdda1e34f02b5979c7884920ae6ba3ad85aa5 does not throw IOException.